src/Sentry.Serilog/SentrySinkExtensions.cs

```diff
@@ -159,130 +205,54 @@ public static class SentrySinkExtensions
         SentryLevel? diagnosticLevel = null,
         ReportAssembliesMode? reportAssembliesMode = null,
         DeduplicateMode? deduplicateMode = null,
-        bool? initializeSdk = null,
         Dictionary<string, string>? defaultTags = null)
     {
-        if (dsn is not null)
-        {
-            sentrySerilogOptions.Dsn = dsn;
-        }
-
-        if (minimumEventLevel.HasValue)
-        {
-            sentrySerilogOptions.MinimumEventLevel = minimumEventLevel.Value;
-        }
-
-        if (minimumBreadcrumbLevel.HasValue)
-        {
-            sentrySerilogOptions.MinimumBreadcrumbLevel = minimumBreadcrumbLevel.Value;
-        }
-
-        if (formatProvider != null)
-        {
-            sentrySerilogOptions.FormatProvider = formatProvider;
-        }
-
-        if (textFormatter != null)
-        {
-            sentrySerilogOptions.TextFormatter = textFormatter;
-        }
-
-        if (sendDefaultPii.HasValue)
-        {
-            sentrySerilogOptions.SendDefaultPii = sendDefaultPii.Value;
-        }
-
-        if (isEnvironmentUser.HasValue)
-        {
-            sentrySerilogOptions.IsEnvironmentUser = isEnvironmentUser.Value;
-        }
-
-        if (!string.IsNullOrWhiteSpace(serverName))
-        {
-            sentrySerilogOptions.ServerName = serverName;
-        }
-
-        if (attachStackTrace.HasValue)
-        {
-            sentrySerilogOptions.AttachStacktrace = attachStackTrace.Value;
-        }
-
-        if (maxBreadcrumbs.HasValue)
-        {
-            sentrySerilogOptions.MaxBreadcrumbs = maxBreadcrumbs.Value;
-        }
-
-        if (sampleRate.HasValue)
-        {
-            sentrySerilogOptions.SampleRate = sampleRate;
-        }
-
-        if (!string.IsNullOrWhiteSpace(release))
-        {
-            sentrySerilogOptions.Release = release;
-        }
-
-        if (!string.IsNullOrWhiteSpace(environment))
-        {
-            sentrySerilogOptions.Environment = environment;
-        }
-
-        if (maxQueueItems.HasValue)
-        {
-            sentrySerilogOptions.MaxQueueItems = maxQueueItems.Value;
-        }
-
-        if (shutdownTimeout.HasValue)
-        {
-            sentrySerilogOptions.ShutdownTimeout = shutdownTimeout.Value;
-        }
+        SetIfNotNull(dsn, value => sentrySerilogOptions.Dsn = value);
```

Comment:
I don't necessarily find this easier to maintain. The original code is easy/simple enough to read, now there's a function call and a lambda and `!`'s everywhere which raise alarm (even though the SetIfNotNull validates it should be safe)

Reply:
All good, let's close this one then.